### ISSUE TITLE
Installation issues

### ISSUE THREAD

**Issue by dandavies99** [symmy596/SurfinPy#23]
Part of [JOSS review](https://github.com/openjournals/joss-reviews/issues/4014) and related to #22 and #20.  

Installation instructions should probably be reviewed. At the moment there are a few interrelated problems:

- `pip install surfinpy` results in a successful installation, but `python setup.py test` fails due to `no module named 'yaml'`. Suggest that `pyyaml` needs adding to `install_requires` and on the version on PyPI. 

- The alternative install instructions (`pip install -r requirements.txt` etc.) don't work on a mac (at least not on my machine - osx 11.5.1 with xcode 13.2.1). This is because of a gcc error (see below) that comes from trying to install `pymatgen==2020.1.28`. There have been [similar issues](https://github.com/materialsproject/pymatgen/issues/1619) around this before. My suggestion would be to pin to a more recent version of pymatgen in requirements.txt. Alternatively, set `pymatgen>=2020.1.28` and `numpy>=1.17.2` if possible. 

```
gcc: error: unrecognized command-line option '-stdlib=libc++'
error: command 'gcc' failed with exit status 1
```

- Again, relating to pymatgen, the package is not actually compatible with python=3.5+ as it says in the README due to `pymatgen==2020.1.28` requirement (see below). Suggest change to python 3.6+ and adding `python_requires='>=3.6',` to `setup()` in `setup.py`. 

```
Could not find a version that satisfies the requirement pymatgen==2020.1.28
```

- In general, if specific versions of every dependency are pinned in requirements.txt, it might be good to add in the README that it is recommended to create a new (conda) environment for this library. Otherwise, the chances of this installing nicely in the same environment of other packages that use matplotlib, numpy, pymatgen etc. are quite low. 

- The local install instructions could probably be simplified. `pip install -r requirements.txt -e .` will install the package in the current directory using `pip` so you get the advantage of pip knowing about it and it's dependencies. `-e` is optional and installs in "editable mode" so any changes made to the code base are applied immediately.

**Comment by awvwgk** [comment]
> * `pip install surfinpy` results in a successful installation, but `python setup.py test` fails due to `no module named 'yaml'`. Suggest that `pyyaml` needs adding to `install_requires` and on the version on PyPI.

Interesting, didn't notice this when installing in a conda env and `pip check` didn't find an issue as well.

**Comment by dandavies99** [comment]
Yeah, that makes sense. Over in [surfaxe](https://github.com/SMTG-UCL/surfaxe) we switched to `ruaml` (e.g. `from ruamel.yaml import YAML`) the same as pymatgen, seeing as it was a dependency anyway.

**Comment by jstse** [comment]
Removed the requirements for specific versions as we do not required specific versions.  Hopefully, this will correct the installation issues you was having.

Updated the local instalation instructions as requested to use `pip install -r requirements.txt -e .`

See Pull request for more information #27
